File: Top Interview 150/C++ Code/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& nums) {
        int ans=0;
        int mini=nums[0];
        for(int i=1;i<nums.size();i++){
            ans=max(ans,nums[i]-mini);
            mini=min(nums[i],mini);
            
        }
        return ans;
    }
};

File: Top Interview 150/Java Code/121-best-time-to-buy-and-sell-stock.java
// Time: O(n)
// Space: O(1)
class Solution {
    public int maxProfit(int[] prices) {
        int ans=0;int mini=prices[0];
        for(int i=1;i<prices.length;i++){
            ans=Math.max(ans,prices[i]-mini);
            mini=Math.min(mini,prices[i]);
        }
        return ans;
    }
}
